DOCSP-45736-add-self-managed-to-glossary (#182)
mongodb/docs-meta: source/style-guide/includes/terms/o.rst

   on-premises
     **off-premises**

>      :icon-fa4:`times-circle` Avoid when referring to non-Atlas
>      deployments. To refer to non-Atlas deployments, use
>      :term:`self-managed`.
> 
>      When used in other contexts, use *on-premises* to distinguish local
>      computing (in which resources are located on a customer's own site)
>      from remote computing (in which resources are provided partially or
>      totally through the cloud). Always hyphenate. *Premises* is plural;

Reviewer: ```suggestion
     :icon-fa4:`times-circle` Avoid this term when you refer to non-Atlas
     deployments. To refer to non-Atlas deployments, use
     :term:`self-managed`.

     When you use this term in other contexts, use *on-premises* to distinguish local
     computing (in which resources are located on a customer's own site)
     from remote computing (in which resources are provided partially or
     totally through the cloud). Always hyphenate. *Premises* is plural;
```

Author: Updated with a few wording tweaks.

   on-premises
     **off-premises**

     :icon-fa4:`times-circle` Avoid using these terms to distinguish
     non-Atlas deployments. To distinguish non-Atlas deployments, use
     :term:`self-managed`.

     Use *on-premises* to distinguish local computing (in which
     resources are located on a customer's own site) from remote
     computing (in which resources are provided partially or totally
     through the cloud). Always hyphenate. *Premises* is plural; don't

>      use *on-premise* or *off-premise*.

Reviewer 2: [q] Not sure how prevalent it is here, but in past companies I've also seen `on-prem` used pretty frequently - is it worth calling that out here or does this cover it?

Author: @stephmarie17 thanks for this comment. I added a small clarification that for consistency we should use "on-premises". Let me know if you'd like to see anything different here.

Reviewer 2: LGTM 👍
